core/src/tile/tileManager.cpp

```diff
@@ -348,27 +349,29 @@ void TileManager::updateTileSet(TileSet& _tileSet, const ViewState& _view) {
             auto& entry = curTilesIt->second;
             entry.setVisible(true);
 
-            auto sourceGeneration = (entry.isReady()) ?
-                entry.tile->sourceGeneration() : entry.task->sourceGeneration();
-
             if (entry.isReady()) {
                 m_tiles.push_back(entry.tile);
 
-                if (!entry.isInProgress() &&
-                    (sourceGeneration < generation)) {
+                auto sourceGeneration = entry.tile->sourceGeneration();
+                if (!entry.isInProgress() && (sourceGeneration < generation)) {
                     // Tile needs update - enqueue for loading
                     entry.task = _tileSet.source->createTask(visTileId);
                     enqueueTask(_tileSet, visTileId, _view);
                 }
             } else if (entry.needsLoading()) {
                 // Not yet available - enqueue for loading
+                if (!entry.task) {
+                    entry.task = _tileSet.source->createTask(visTileId);
```

Comment:
Can we put a comment here, mentioning how this is possible?
Basically, an entry's task could be null either when its ready (implying tile is loaded) or when its cleared (cancelled from loading and cleared, or removed from tileSet completely).

Here it seems to be because of https://github.com/tangrams/tangram-es/blob/4de407f92284649cadc4def9c5544c4a0ece0cb8/core/src/tile/tileManager.cpp#L418

Reply:
If https://github.com/tangrams/tangram-es/pull/1929/files#r232113259 is acknowledged then I do not think we need to create a task here, since the entry will itself be removed.